spec.emu

```diff
@@ -5,447 +5,144 @@
-            <p><ins>This List respects the following invariants:</ins></p>
-            <ul>
-              <li>
-                <ins>It does not contain two Records _record1_ and _record2_ such that _record1_.[[ReferencingScriptOrModule]] is _record2_.[[ReferencingScriptOrModule]] and _record1_.[[Specifier]] is _record2_.[[Specifier]].</ins>
-              </li>
-              <li>
-                <p><ins>For every Record _record_ it contains, if Type(_record_.[[Specifier]]) is Object then:</ins></p>
-                <ul>
-                  <li>
-                    <ins>_record_.[[Specifier]] has a [[ModuleBlockBody]] internal slot.</ins>
-                  </li>
-                  <li>
-                    <ins>_record_.[[ReferencingScriptOrModule]] is *null*.</ins>
-                  </li>
-                </ul>
-              </li>
-            </ul>
-          </td>
-        </tr>
-        <tr>
-          <td>
-            [[HostDefined]]
-          </td>
-          <td></td>
-          <td></td>
-        </tr>
-      </table>
-    </emu-table>
+<emu-note type="editor">
+  This document is a diff on top of&nbsp;<a href="https://github.com/tc39/ecma262/pull/2905">tc39/ecma262#2905</a>.
+</emu-note>
+
+<emu-clause id="sec-ecmascript-language-expressions" number="13">
+  <h1>ECMAScript Language: Expressions</h1>
+
+  <emu-clause id="sec-primary-expression" number="2">
+    <h1>Primary Expression</h1>
+    <h2>Syntax</h2>
+    <emu-grammar type="definition">
+      PrimaryExpression[Yield, Await] :
+        `this`
+        IdentifierReference[?Yield, ?Await]
+        Literal
+        ArrayLiteral[?Yield, ?Await]
+        ObjectLiteral[?Yield, ?Await]
+        FunctionExpression
+        ClassExpression[?Yield, ?Await]
+        GeneratorExpression
+        AsyncFunctionExpression
+        AsyncGeneratorExpression
+        <ins>ModuleBlockExpression</ins>
+        RegularExpressionLiteral
+        TemplateLiteral[?Yield, ?Await, ~Tagged]
+        CoverParenthesizedExpressionAndArrowParameterList[?Yield, ?Await]
+    </emu-grammar>
+
+    <emu-clause id="sec-module-block-expression" number="10">
+      <h1><ins>Module Block Expression</ins></h1>
+      <h2>Syntax</h2>
+      <emu-grammar type="definition">
+        ModuleBlockExpression :
+          `module` [no LineTerminator here] `{` ModuleBody? `}`
+      </emu-grammar>
+
+      <emu-clause id="sec-module-block-expression-runtime-semantics-evaluation" type="sdo">
+        <h1>Runtime Semantics: Evaluation</h1>
+        <emu-grammar>ModuleBlockExpression: `module` `{` ModuleBody `}`</emu-grammar>
+        <emu-alg>
+          1. Let _moduleBlock_ be OrdinaryObjectCreate(%ModuleBlock.prototype%, &laquo; [[ModuleRecord]], [[SourceText]] &raquo;).
```

Comment:
don't we use the %Module.prototype% from https://tc39.es/proposal-compartments/0-module-and-module-source.html#sec-module-constructor?

Reply:
Yes, the proposal just needs to be made compatible with the compartments one. I plan to do as soon as this is merged.